spec/lib/pdf_filler/nj1040_pdf_spec.rb

```diff
@@ -1464,6 +1446,25 @@ def map_spouse_death_year(input)
         expect(pdf_fields["Text40"]).to eq "0"
         expect(pdf_fields["Text41"]).to eq "0"
       end
+
+      it "leaves blank when no NewJerseyTaxableIncome" do
+        allow_any_instance_of(Efile::Nj::Nj1040Calculator).to receive(:calculate_line_42).and_return 0
+        # millions
+        expect(pdf_fields["Enter Code4332"]).to eq ""
+        expect(pdf_fields["40"]).to eq ""
+        expect(pdf_fields["undefined_114"]).to eq ""
+        # thousands
+        expect(pdf_fields["Text19"]).to eq ""
+        expect(pdf_fields["Text20"]).to eq ""
+        expect(pdf_fields["Text30"]).to eq ""
+        # hundreds
+        expect(pdf_fields["Text37"]).to eq ""
+        expect(pdf_fields["Text38"]).to eq ""
+        expect(pdf_fields["Text39"]).to eq ""
+        # decimals
+        expect(pdf_fields["Text40"]).to eq ""
+        expect(pdf_fields["Text41"]).to eq ""
+      end
```

Comment:
i added one test here because there wasn't yet one for the case where this is blank